libsolidity/ast/Types.cpp

```diff
@@ -613,6 +613,19 @@ TypeResult IntegerType::binaryOperatorResult(Token _operator, Type const* _other
 		}
 		else if (dynamic_cast<FixedPointType const*>(_other))
 			return nullptr;
+		else if (
+			auto rationalNumberType = dynamic_cast<RationalNumberType const*>(_other);
+			rationalNumberType
+		)
+		{
+			IntegerType const* integerType = rationalNumberType->integerType();
+			if (
+				!integerType ||
+				integerType->numBits() >
```

Comment:
I think the first condition is enough, at least the second condition is not grounded on anything. For non-literals, we make no restrictions on the number of bits, for example.

Reply:
@chriseth Don't you think code like this should be dissallowed?
```solidity
contract C {
	function f(uint8 x) public pure {
		x = x ** 1E5;
	}
}
```

Reply:
I'm just saying we should be consistent.